libs/sdk-core/src/node_api.rs

```diff
@@ -46,6 +46,18 @@ pub enum NodeError {
 
     #[error("Service connectivity: {0}")]
     ServiceConnectivity(anyhow::Error),
+
+    #[error("Invalid offer: {0:?}")]
+    InvalidOffer(anyhow::Error),
+
+    #[error("Offer expired: {0}")]
+    OfferExpired(anyhow::Error),
+
+    #[error("Offer reply error: {0}")]
+    OfferReplyError(anyhow::Error),
+
+    #[error("Offer timeout: {0}")]
+    OfferTimeout(anyhow::Error),
```

Comment:
Suggestion to make more clear what this does.
It's a long name I'm suggesting though, maybe we can do better.
```suggestion
    OfferInvoiceRequestTimeout(anyhow::Error),
```